pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -23,17 +27,31 @@ type executionStartInfo struct {
 	startTime time.Time
 	traceID   uint64
 	spanID    uint64
+	parentID  uint64
+}
+type invocationPayload struct {
+	Headers map[string]string `json:"headers"`
 }
 
 // currentExecutionInfo represents information from the start of the current execution span
 var currentExecutionInfo executionStartInfo
 
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func startExecutionSpan(startTime time.Time) {
+func startExecutionSpan(startTime time.Time, rawPayload string) {
 	currentExecutionInfo.startTime = startTime
 	currentExecutionInfo.traceID = random.Uint64()
 	currentExecutionInfo.spanID = random.Uint64()
+
+	payload, hasPayload := convertRawPayload(rawPayload)
+
+	if hasPayload {
```

Comment:
What happens if there *is* a payload, but the payload doesn't contain any trace headers? For example, the customer isn't using the Datadog tracer in the upstream service. Can we add a unit test to verify this behavior as well?